File: frontend-web/src/types/branch.ts
export interface Branch {
  id: number;
  name: string;
  code: string;
  address: string;
  commune: string;
  department: string;
  phones: string[];
  email: string;
  openingDate: string;
  managerId?: string;
  managerName?: string;
  maxEmployees: number;
  status: BranchStatus;
  limits: BranchLimits;
  operatingHours: OperatingHours;
  createdAt: string;
  updatedAt: string;
}

export enum BranchStatus {
  Active = 'Active',
  Inactive = 'Inactive',
  UnderConstruction = 'UnderConstruction'
}

export interface BranchLimits {
  dailyWithdrawalLimit: number;
  dailyDepositLimit: number;
  maxLocalCreditApproval: number;
  minCashReserveHTG: number;
  minCashReserveUSD: number;
}

export interface OperatingHours {
  openTime: string;
  closeTime: string;
  closedDays: DayOfWeek[];
}

export enum DayOfWeek {
  Sunday = 0,
  Monday = 1,
  Tuesday = 2,
  Wednesday = 3,
  Thursday = 4,
  Friday = 5,
  Saturday = 6
}

export interface CreateBranchRequest {
  name: string;
  code?: string;
  address: string;
  commune: string;
  department: string;
  phones: string[];
  email: string;
  openingDate: string;
  managerId?: string;
  maxEmployees: number;
  status: BranchStatus;
  limits: BranchLimits;
  operatingHours: OperatingHours;
}

export interface UpdateBranchRequest extends Partial<CreateBranchRequest> {
  id: number;
}

export interface BranchFormData {
  name: string;
  code: string;
  address: string;
  commune: string;
  department: string;
  phone1: string;
  phone2?: string;
  phone3?: string;
  email: string;
  openingDate: string;
  managerId?: string;
  maxEmployees: number;
  status: BranchStatus;
  
  // Limits
  dailyWithdrawalLimit: number;
  dailyDepositLimit: number;
  maxLocalCreditApproval: number;
  minCashReserveHTG: number;
  minCashReserveUSD: number;
  
  // Operating Hours
  openTime: string;
  closeTime: string;
  closedDays?: DayOfWeek[];
}

export interface BranchHistory {
  id: number;
  branchId: number;
  action: string;
  description: string;
  performedBy: string;
  performedAt: string;
  oldValue?: any;
  newValue?: any;
}

// Haitian Departments
export const HAITI_DEPARTMENTS = [
  'Artibonite',
  'Centre',
  'Grand\'Anse',
  'Nippes',
  'Nord',
  'Nord-Est',
  'Nord-Ouest',
  'Ouest',
  'Sud',
  'Sud-Est'
] as const;

// Common communes by department
export const COMMUNES_BY_DEPARTMENT: Record<string, string[]> = {
  'Ouest': [
    'Port-au-Prince',
    'Carrefour',
    'Delmas',
    'Pétion-Ville',
    'Tabarre',
    'Cité Soleil',
    'Kenscoff',
    'Gressier',
    'Léogâne',
    'Grand-Goâve',
    'Petit-Goâve',
    'Arcahaie',
    'Cabaret',
    'Cornillon',
    'Croix-des-Bouquets'
  ],
  'Artibonite': [
    'Gonaïves',
    'Saint-Marc',
    'Dessalines',
    'Grande-Saline',
    'L\'Estère',
    'Petite-Rivière-de-l\'Artibonite',
    'Verrettes'
  ],
  'Nord': [
    'Cap-Haïtien',
    'Fort-Dauphin',
    'Grande-Rivière-du-Nord',
    'Quartier-Morin',
    'Limonade',
    'Plaine-du-Nord',
    'Dondon',
    'Saint-Raphaël'
  ],
  'Sud': [
    'Les Cayes',
    'Aquin',
    'Saint-Louis-du-Sud',
    'Cavaillon',
    'Port-Salut',
    'Roche-à-Bateau',
    'Torbeck'
  ],
  'Centre': [
    'Hinche',
    'Mirebalais',
    'Lascahobas',
    'Belladère',
    'Savanette',
    'Cerca-la-Source',
    'Thomonde'
  ],
  'Grand\'Anse': [
    'Jérémie',
    'Abricots',
    'Anse-d\'Hainault',
    'Corail',
    'Pestel',
    'Roseaux',
    'Beaumont'
  ],
  'Nippes': [
    'Miragoâne',
    'Petit-Trou-de-Nippes',
    'Anse-à-Veau',
    'Arnaud',
    'Baradères',
    'Fond-des-Nègres',
    'Plaisance-du-Sud'
  ],
  'Nord-Est': [
    'Fort-Liberté',
    'Ouanaminthe',
    'Terrier-Rouge',
    'Capotille',
    'Mont-Organisé',
    'Ferrier'
  ],
  'Nord-Ouest': [
    'Port-de-Paix',
    'Jean-Rabel',
    'Môle-Saint-Nicolas',
    'Bassin-Bleu',
    'Bombardopolis',
    'Chansolme'
  ],
  'Sud-Est': [
    'Jacmel',
    'Cayes-Jacmel',
    'Marigot',
    'Bainet',
    'Côtes-de-Fer',
    'La Vallée',
    'Belle-Anse'
  ]
};

File: backend/NalaCreditAPI/Models/Branch.cs
using System.ComponentModel.DataAnnotations;

namespace NalaCreditAPI.Models;

public class Branch
{
    public int Id { get; set; }

    [Required]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(20)]
    public string? Code { get; set; }

    [Required]
    [MaxLength(200)]
    public string Address { get; set; } = string.Empty;

    [MaxLength(100)]
    public string? Commune { get; set; }

    [MaxLength(50)]
    public string? Department { get; set; }

    public List<string> Phones { get; set; } = new List<string>();

    [MaxLength(100)]
    public string? Email { get; set; }

    public DateTime OpeningDate { get; set; } = DateTime.UtcNow;

    public string? ManagerId { get; set; }

    public string? ManagerName { get; set; }

    public int MaxEmployees { get; set; } = 50;

    [Required]
    public string Region { get; set; } = string.Empty;

    [Required]
    public Currency PrimaryCurrency { get; set; }

    public bool AcceptsUSD { get; set; } = true;
    public bool AcceptsHTG { get; set; } = true;

    public decimal DailyTransactionLimit { get; set; }
    public decimal CashLimit { get; set; }

    // Operating Hours
    public TimeSpan OpenTime { get; set; } = new TimeSpan(8, 0, 0); // 8:00 AM
    public TimeSpan CloseTime { get; set; } = new TimeSpan(17, 0, 0); // 5:00 PM
    public List<int> ClosedDays { get; set; } = new List<int>(); // 0=Sunday, 1=Monday, etc.

    // Branch Limits
    public decimal DailyWithdrawalLimit { get; set; } = 1000000;
    public decimal DailyDepositLimit { get; set; } = 1000000;
    public decimal MaxLocalCreditApproval { get; set; } = 500000;
    public decimal MinCashReserveHTG { get; set; } = 50000;
    public decimal MinCashReserveUSD { get; set; } = 1000;

    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public User? Manager { get; set; }
    public ICollection<User> Users { get; set; } = new List<User>();
    public ICollection<Account> Accounts { get; set; } = new List<Account>();
    public ICollection<Transaction> Transactions { get; set; } = new List<Transaction>();
    public ICollection<CashSession> CashSessions { get; set; } = new List<CashSession>();
    public ICollection<InterBranchTransfer> SentTransfers { get; set; } = new List<InterBranchTransfer>();
    public ICollection<InterBranchTransfer> ReceivedTransfers { get; set; } = new List<InterBranchTransfer>();
}

public enum Currency
{
    HTG = 1,
    USD = 2
}